packages/itwin/measure-tools/src/measurements/PerpendicularDistanceMeasurement.ts

```diff
@@ -0,0 +1,237 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Point3d, type XYAndZ, type XYZProps } from "@itwin/core-geometry";
+import { ColorDef, LinePixels } from "@itwin/core-common";
+import type { DecorateContext } from "@itwin/core-frontend";
+import { GraphicType, IModelApp, QuantityType } from "@itwin/core-frontend";
+import { Measurement, type MeasurementEqualityOptions, type MeasurementWidgetData } from "../api/Measurement";
+import { MeasureTools } from "../MeasureTools";
+import { DistanceMeasurement, type DistanceMeasurementProps, DistanceMeasurementSerializer } from "./DistanceMeasurement";
+import type { MeasurementProps } from "../api/MeasurementProps";
+
+export enum PerpendicularMeasurementType {
+  Width = "width",
+  Height = "height",
+}
+
+/**
+ * Props for serializing a [[PerpendicularDistanceMeasurement]].
+ */
+export interface PerpendicularDistanceMeasurementProps extends DistanceMeasurementProps {
+  measurementType: PerpendicularMeasurementType;
+  secondaryLine?: XYZProps[];
+}
+
+/** Serializer for a [[PerpendicularDistanceMeasurement]]. */
+export class PerpendicularDistanceMeasurementSerializer extends DistanceMeasurementSerializer {
+  public static readonly perpendicularDistanceMeasurementName = "perpendicularDistanceMeasurement";
+
+  public override get measurementName(): string {
+    return PerpendicularDistanceMeasurementSerializer.perpendicularDistanceMeasurementName;
+  }
+
+  public override isValidType(measurement: Measurement): boolean {
+    return measurement instanceof PerpendicularDistanceMeasurement;
+  }
+
+  public override isValidJSON(json: any): boolean {
+    if (!super.isValidJSON(json) || !json.hasOwnProperty("startPoint") || !json.hasOwnProperty("endPoint") || !json.hasOwnProperty("measurementType")) {
```

Comment:
@saaaaaally Can omit the startPoint/endPoint check since that will be taken care of in the super call.